libzmix/src/signatures/delg_cred_cdd/README.md

```diff
@@ -0,0 +1,46 @@
+[Practical UC-Secure Delegatable Credentials with Attributes and Their Application to Blockchain](https://acmccs.github.io/papers/p683-camenischA.pdf)
+
+**This implementation is not UC-secure but only implements the delegatable credential part**
+
+### Brief description of the API
+1. [Groth1 and Groth2 signatures](src/groth_sig.rs).  
+    - Parameters can be generated by calling `GrothS1::setup` or `GrothS2::setup`. `setup` takes the maximum number of attributes that need to be supported. Keep it one more than the number you want to support to accommodate the public key.
+    - Signing keys can be generated by calling `GrothS1::keygen` or `GrothS2::keygen`. Takes the corresponding setup parameters.
+    - A new signature can be created by calling `Groth1Sig:new` or `Groth2Sig:new`. An existing signature can be randomized by calling `randomize` on the siganture.
+    - 2 methods for signature verification, `verify` and `verify_fast`, both with the same API. `verify` computes several pairings to verify the signature whereas `verify_fast` does only 1 big multi-pairing. Applies this observation to pairings: if it needs to be cheched that a == b and c == d and e == f, then choose a random number `r` and check whether (a-b) + (c-d)*r + (e-f)*r<sup>2</sup> == 0. Refer the docs for the method for more details     
+2. [Issuers and delegation](src/issuer.rs).  
+   - Issuers are instantiated by calling `EvenLevelIssuer::new` or `OddLevelIssuer::new` by passing their level to the `new` function. Root issuers is at level 0 so always instantiated by `EvenLevelIssuer::new(0)`.
+   - Issuers generate their keys with `EvenLevelIssuer::keygen` or `OddLevelIssuer::keygen`.
+   - Issuers can delegate by calling `delegate` method that takes the attributes to sign, who to delegate to etc resulting in a credential. 
```

Comment:
Issuers delegate by calling the `delegate` method which takes the attributes to sign and to whom to delegate. The output is the new credential.